### ISSUE TITLE
python3Packages.ipykernel fails to build

### ISSUE THREAD

**Issue by marcus7070** [NixOS/nixpkgs#87166]
**Describe the bug**
`python3Packages.ipykernel` fails to build. Tests fail due to change in `python3Packages.jupyter_client` here: e19e5ac7f00faba7bbfee0d6ba30d24ccb62e584 

**To Reproduce**
build with any commit in the master branch after e19e5ac7f00faba7bbfee0d6ba30d24ccb62e584.

**Expected behavior**
Successful build of the python packages `ipykernel`, `spyder_3`, and the top level attribute `cq-editor`.

**Logs**
```
> nix-build -A python3Packages.ipykernel                                                                
these derivations will be built:
  /nix/store/j4p4x311qrn1nyygbpr67zqj3bzaq3cr-python3.7-ipykernel-5.1.4.drv
building '/nix/store/j4p4x311qrn1nyygbpr67zqj3bzaq3cr-python3.7-ipykernel-5.1.4.drv'...
<snip>
running install tests
no Makefile or custom buildPhase, doing nothing
pythonCatchConflictsPhase
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
pytestCheckPhase
Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.7.7, pytest-5.3.5, py-1.8.1, pluggy-0.13.1
rootdir: /build/ipykernel-5.1.4
plugins: flaky-3.6.1
collected 107 items                                                            

ipykernel/inprocess/tests/test_kernel.py FFFF                            [  3%]
ipykernel/inprocess/tests/test_kernelmanager.py FFFFF                    [  8%]
ipykernel/tests/test_async.py ..ss                                       [ 12%]
ipykernel/tests/test_connect.py .....                                    [ 16%]
ipykernel/tests/test_embed_kernel.py ...                                 [ 19%]
ipykernel/tests/test_eventloop.py .                                      [ 20%]
ipykernel/tests/test_heartbeat.py ....                                   [ 24%]
ipykernel/tests/test_io.py .                                             [ 25%]
ipykernel/tests/test_jsonutil.py .....                                   [ 29%]
ipykernel/tests/test_kernel.py .......s.....s..                          [ 44%]
ipykernel/tests/test_kernelspec.py ..........                            [ 54%]
ipykernel/tests/test_message_spec.py ......................              [ 74%]
ipykernel/tests/test_pickleutil.py .....                                 [ 79%]
ipykernel/tests/test_serialize.py ....ssss......                         [ 92%]
ipykernel/tests/test_start_kernel.py ..                                  [ 94%]
ipykernel/tests/test_zmq_shell.py ......                                 [100%]

=================================== FAILURES ===================================
_________________ InProcessKernelTestCase.test_getpass_stream __________________

self = <ipykernel.inprocess.tests.test_kernel.InProcessKernelTestCase testMethod=test_getpass_stream>

    def setUp(self):
        self.km = InProcessKernelManager()
        self.km.start_kernel()
        self.kc = self.km.client()
>       self.kc.start_channels()

ipykernel/inprocess/tests/test_kernel.py:29:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
ipykernel/inprocess/client.py:64: in start_channels
    super(InProcessKernelClient, self).start_channels()
/nix/store/fn1cz5ralsv3437lryfng7wibkvsqa1n-python3.7-jupyter_client-6.1.3/lib/python3.7/site-packages/jupyter_client/client.py:118: in start_channels
    self.control_channel.start()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <ipykernel.inprocess.blocking.BlockingInProcessKernelClient object at 0x7ffff496b8d0>

    @property
    def control_channel(self):
        """Get the control channel object for this kernel."""
        if self._control_channel is None:
            url = self._make_url('control')
            self.log.debug("connecting control channel to %s", url)
            socket = self.connect_control(identity=self.session.bsession)
            self._control_channel = self.control_channel_class(
>               socket, self.session, self.ioloop
            )
E           TypeError: ChannelABC() takes no arguments

/nix/store/fn1cz5ralsv3437lryfng7wibkvsqa1n-python3.7-jupyter_client-6.1.3/lib/python3.7/site-packages/jupyter_client/client.py:200: TypeError
______________________ InProcessKernelTestCase.test_pylab ______________________
ipykernel/inprocess/tests/test_kernel.py:29: in setUp
    self.kc.start_channels()
ipykernel/inprocess/client.py:64: in start_channels
    super(InProcessKernelClient, self).start_channels()
/nix/store/fn1cz5ralsv3437lryfng7wibkvsqa1n-python3.7-jupyter_client-6.1.3/lib/python3.7/site-packages/jupyter_client/client.py:118: in start_channels
    self.control_channel.start()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <ipykernel.inprocess.blocking.BlockingInProcessKernelClient object at 0x7ffff4923a10>

    @property
    def control_channel(self):
        """Get the control channel object for this kernel."""
        if self._control_channel is None:
            url = self._make_url('control')
            self.log.debug("connecting control channel to %s", url)
            socket = self.connect_control(identity=self.session.bsession)
            self._control_channel = self.control_channel_class(
>               socket, self.session, self.ioloop
            )
E           TypeError: ChannelABC() takes no arguments

/nix/store/fn1cz5ralsv3437lryfng7wibkvsqa1n-python3.7-jupyter_client-6.1.3/lib/python3.7/site-packages/jupyter_client/client.py:200: TypeError
____________________ InProcessKernelTestCase.test_raw_input ____________________
<snipped many more errors, all in jupyter_client>
```

**Extra context**
`ipykernel` in nixpkgs is 5.1.4, they've released 5.2.1, perhaps a version bump will fix? https://github.com/ipython/ipykernel/releases

**Notify maintainers**

Maintainer information:
@FRidh 
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  python3Packages.ipykernel
  python3Packages.spyder_kernels
  python3Packages.spyder
  python3Packages.spyder_3
  cq-editor
  python3Packages.bash_kernel
  etc.
module:
```

**Comment by FRidh** [comment]
I noticed this yesterday as well. Also need to consider a pyzmq bump.

**Comment by marcus7070** [comment]
Closed by #87223
